packages/rum-recorder/src/domain/deflateSegmentWriter.ts

```diff
@@ -1,54 +1,44 @@
-import { addMonitoringMessage, addErrorToMonitoringBatch, monitor } from '@datadog/browser-core'
-import { SegmentMeta } from '../types'
-import { DeflateWorker } from './deflateWorker'
+import { addMonitoringMessage, monitor } from '@datadog/browser-core'
+import { DeflateWorker, DeflateWorkerListener } from './deflateWorker'
 import { SegmentWriter } from './segment'
 
-export class DeflateSegmentWriter implements SegmentWriter {
-  private nextId = 0
-  private pendingMeta: Array<{ id: number; meta: SegmentMeta }> = []
+let nextId = 0
 
-  constructor(
-    private worker: DeflateWorker,
-    private onWrote: (size: number) => void,
-    private onFlushed: (data: Uint8Array, meta: SegmentMeta) => void
-  ) {
-    worker.addEventListener(
-      'message',
-      monitor(({ data }) => {
-        if ('error' in data) {
-          addErrorToMonitoringBatch(data.error)
-        } else if ('result' in data) {
-          let pendingMeta = this.pendingMeta.shift()!
+export class DeflateSegmentWriter implements SegmentWriter {
+  private id = nextId++
 
-          // Messages should be received in the same order as they are sent, so the first
-          // 'pendingMeta' of the list should be the one corresponding to the handled message.
-          // But if something goes wrong in the worker and a response is lost, we need to avoid
-          // associating an incorrect meta to the flushed segment. Remove any pending meta with an id
-          // inferior to the one being waited for.
-          if (pendingMeta.id !== data.id) {
-            let lostCount = 0
-            while (pendingMeta.id !== data.id) {
-              pendingMeta = this.pendingMeta.shift()!
-              lostCount += 1
-            }
-            addMonitoringMessage(`${lostCount} deflate worker responses have been lost`)
+  constructor(private worker: DeflateWorker, onWrote: (size: number) => void, onFlushed: (data: Uint8Array) => void) {
+    const listener: DeflateWorkerListener = monitor(({ data }) => {
+      if (!('error' in data)) {
+        if (data.id > this.id) {
+          // Messages should be received in the same order as they are sent, so if we receive a
+          // message with an id superior to this DeflateSegmentWriter instance id, we know that
```

Comment:
I am not sure to see what are the pros/cons of having separate writer instances compared to keep track of sent messages in the writer to prevent flushing.
Any insight?